### PR TITLE
Metrics grammar updates

### PR DESCRIPTION
**Pull request by efronbs**
~TODO: write changelog~
issue: https://github.com/yahoo/fili/issues/1054

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/RegexApiFilterParser.java  8
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/FilterDefinition.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5528246&bid=18119510)

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/FilterDefinition.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    public void setFieldName(String fieldName) {
+        this.fieldName = fieldName;
+    }
+
+    public String getFieldName() {
+        return fieldName;
+    }
+
+    public void setOperationName(String operationName) {
```
Comment (michael-mclawhorn) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/FilterDefinition.java:62]: Pull out setters.

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/RegexApiFilterParser.java  8
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/FilterDefinition.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5528246&bid=18119510)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/FilterDefinition.java  7
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/RegexApiFilterParser.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5528246&bid=18119510)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/FilterDefinition.java  7
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/RegexApiFilterParser.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5528246&bid=18119510)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/RegexApiFilterParser.java  8
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/metric/ProtocolLogicalMetricGenerator.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/filter/FilterDefinition.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5528246&bid=18119510)

**Pull request merged**